java-client-model/src/main/java/com/cumulocity/model/ID.java

```diff
@@ -0,0 +1,118 @@
+package com.cumulocity.model;
+
+import static com.cumulocity.model.util.ExtensibilityConverter.classToStringRepresentation;
+
+import org.svenson.AbstractDynamicProperties;
+import org.svenson.JSONProperty;
+
+public class ID extends AbstractDynamicProperties {
```

Comment:
I would check if this can be skipped, and preserve only GId class.

Reply:
It is used extensively - more then 200 usages in agents and we also don't know how many clients used it, in my opinion this would be to big breaking change.